pom.xml

```diff
@@ -232,6 +232,34 @@
     </build>
 
     <dependencies>
+        <!-- Used by FeatureSetDependenciesMojo -->
+        <dependency>
+            <groupId>org.codehaus.plexus</groupId>
+            <artifactId>plexus-archiver</artifactId>
+            <version>4.9.2</version>
+        </dependency>
+
+        <!-- 4.0.0 is not compatible with Maven 3.9.x -->
+        <dependency>
+            <groupId>org.codehaus.plexus</groupId>
+            <artifactId>plexus-utils</artifactId>
+            <version>3.5.1</version>
```

Comment:
Can we try with https://github.com/codehaus-plexus/plexus-utils/releases/tag/plexus-utils-4.0.1? It has
- https://github.com/codehaus-plexus/plexus-utils/pull/263

Reply:
Wow, right on time! Unfortunately version 4 removed the `org.codehaus.plexus.util.xml` package which is now in `maven-shared-utils` with different api, which is incompatible with the original implementation. Perhaps we can reimplement part of the plugin to respect new relations between dependencies for 4.1.0, now I would just make this smaller step.

Reply:
Just for the record:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:compile (default-compile) on project glassfishbuild-maven-plugin: Compilation failure: Compilation failure:
[ERROR] /media/data-samsung/data/dmatej/work/repo/git/glassfishbuild-maven-plugin/src/main/java/org/glassfish/build/LifecyclePhaseHelper.java:[26,36] package org.codehaus.plexus.util.xml does not exist
[ERROR] /media/data-samsung/data/dmatej/work/repo/git/glassfishbuild-maven-plugin/src/main/java/org/glassfish/build/xpp3dom/ConfigurationElement.java:[20,36] package org.codehaus.plexus.util.xml does not exist
[ERROR] /media/data-samsung/data/dmatej/work/repo/git/glassfishbuild-maven-plugin/src/main/java/org/glassfish/build/xpp3dom/ConfigurationElement.java:[25,49] cannot find symbol
[ERROR]   symbol: class Xpp3Dom
[ERROR] /media/data-samsung/data/dmatej/work/repo/git/glassfishbuild-maven-plugin/src/main/java/org/glassfish/build/LifecyclePhaseHelper.java:[64,12] cannot find symbol
[ERROR]   symbol:   class Xpp3Dom
[ERROR]   location: class org.glassfish.build.LifecyclePhaseHelper
[ERROR] /media/data-samsung/data/dmatej/work/repo/git/glassfishbuild-maven-plugin/src/main/java/org/glassfish/build/xpp3dom/AssemblyDescriptorElements.java:[19,36] package org.codehaus.plexus.util.xml does not exist
[ERROR] /media/data-samsung/data/dmatej/work/repo/git/glassfishbuild-maven-plugin/src/main/java/org/glassfish/build/xpp3dom/AssemblyDescriptorElements.java:[24,49] cannot find symbol
[ERROR]   symbol: class Xpp3Dom
[ERROR] /media/data-samsung/data/dmatej/work/repo/git/glassfishbuild-maven-plugin/src/main/java/org/glassfish/build/xpp3dom/PropertyElement.java:[19,36] package org.codehaus.plexus.util.xml does not exist
[ERROR] /media/data-samsung/data/dmatej/work/repo/git/glassfishbuild-maven-plugin/src/main/java/org/glassfish/build/xpp3dom/PropertyElement.java:[24,44] cannot find symbol
[ERROR]   symbol: class Xpp3Dom
```

Reply:
I gave up, because the plugin uses `org.apache.maven.lifecycle.mapping.LifecycleMojo` from `maven-core 3.9.6`, which imports **Plexus Utils** `import org.codehaus.plexus.util.xml.Xpp3Dom;` , so there is no way using `maven-shared-utils`.

For now I created milestone 5.0.0 which should be compatible with Maven 4 in the future, which probably will have these expected dependencies. 

Maybe there is yet another way, but I don't see it now.